eos-reclaim-swap

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import parted
+import subprocess
+
+class SwapRemover():
+    '''
+    Remove swap partitions from a disk and reclaim their space.
+    '''
+
+    def get_table_str(self, disk):
+        '''
+        Return a text string with the disk partition table.
+        '''
+        cmd = ['fdisk', '-l', disk]
+        out = subprocess.run(cmd, stdout=subprocess.PIPE).stdout
+        return out.decode()
+
+
+    def find_disk(self, partition):
+        '''
+        Return path to a block device representing the disk containing
+        partition.
+        '''
+        p = os.path.realpath(partition)
+        disk = None
+
+        # /dev/sd[a-z][0-9]
+        if (p[:-2] == '/dev/sd'):
+            disk = p[:-1]
+            self.sep = ''
+
+        # /dev/[a-z]+p[0-9], but not /dev/sdp[0-9] (eMMC)
+        elif (p.startswith('/dev/') and p[-2:-1] == 'p'):
```

Comment:
I feel like there must be a better way to navigate from a partition to its host device than by ad-hoc rules about whether there's a `p` in here. This one, for example, would fail if you had more than 9 partitions and the root partition was the 10th. I appreciate that we don't do that!

Reply:
Hum, I never realized negative indexes worked like that outside of range indexes, I'll change it.

I'm aware this assumes at most 9 partitions in a layout, but this restriction is also in endless-repartition.sh which creates the final layouts in the first place. I would also prefer a better way to find the disk, but I don't know any other way from the top of my head and for now this only needs to work with the layouts we have shipped so far. So maybe we could leave this known restriction in place for now and improve it later?